kep/640-error-pods-rescheduling/README.md

```diff
@@ -0,0 +1,166 @@
+such as NUMA or device-related information. In such cases, we need to manually reschedule the pod to a different node. But this pod may schedule to the same node with the default scheduler.
+
+## Design Details
+
+Our scheme is to design two components as outlined in the diagram and described below: Rescheduler-controller and Podrescheduling. 
+Users can configure the `scheduling-retries` (maximum rescheduling retry count) in the metadata annotations of workload resources, such as Deployment and StatefulSet. 
+Rescheduler-controller is responsible for monitoring pods with abnormal states in the cluster. For each abnormal pod, 
+the rescheduler-controller checks the current scheduling count (defined as `currentReschedulingTimes`) recorded in the metadata annotations of its associated workload resource. 
+If the current scheduling count does not exceed the user-configured `scheduling-retries` (maximum rescheduling retry count), 
+it increments the current scheduling count by 1 and records the current node where the pod is located in the scheduled nodes list (eg: defined as `deployScheduledHosts`). 
+Then, rescheduler-controller deletes the abnormal pod, and this pod's associated workload resource will create a new pod automatically. 
+Rescheduler-controller registers a MutateWebHook in the kube-apiserver, this MutateWebHook will patch scheduled nodes list info to this newly created pod's annotations (defined as `sigs.k8s.io/scheduled-hosts`) when this newly created pod comes to kube-apiserver admission point.  
+Podrescheduling is the scheduler-plugins that implement the PreFilter extension point of kube-scheduler to prefilter out the scheduled nodes list recorded in this pod annotation and schedule this pod to other nodes.
+
+![design](./design.png)
+
+### Rescheduler-controller
+
+The rescheduler-controller is comprised of two components, as illustrated in the above diagram: ListFunc and MutateWebHook.
+
+The ListFunc serves as a periodic monitoring program responsible for listing pods in abnormal states within the cluster. 
+It is also in charge of deleting these abnormal pods and updating annotations in the metadata of these pods' workload resources to record the scheduled nodes list. 
+The ListFunc triggers pod rescheduling by deleting pods. Additionally, for pods without an owner reference definition, the ListFunc maintains a map (for updating pod's metada annotations) in memory to store scheduled hosts, allowing for direct code-based resubmission of these pods to the cluster.
+
+The MutateWebHook registers with the kube-apiserver and intercepts incoming pods within the cluster. Subsequently, the MutateWebHook patches the list of scheduled nodes for these pods, which is retrieved from the annotations of the pods' associated workload resources, into the metada annotations of the pods themselves.
```

Comment:
What if we directly impose additional podAntiAffinity field to the ownerReference's podTemplate / orphan pod's spec? In this case, we can simplify the entire workflow and simplify the design to be a controller-only solution.

Reply:
> What if we directly impose additional podAntiAffinity field to the ownerReference's podTemplate / orphan pod's spec? 
This can result in pod's ownerReference

This will cause changes to the pod's `ownerReference` definition, but how does this additional podAntiAffinity field prevent the pod from being scheduled to the previously erroneous node?